JesusGonzalezA/Trabe-Interview
# Feature modal

With this pr, a new component would be added to the repository. This component consists on a **modal**, to represent information above other content on the page.

As the main goal of the project is to create this component, the readme contains all the necessary information to start using it. Nevertheless, I include the description of the props here to make your review job easier! 

How to run the app/test suite
---
Execute the application
```bash
npm i
npm start
```

Execute the test suite
```bash
npm i
npm test
```

How to close the modal
---
The modal can be closed:
- Using a parent component that controls its state
- Using a close button that is included on the modal
- By clicking outside of the modal. 
To check if the development of these features is correct, I included a very simple test suite.

If you need me to improve the code or develop something else, I'm looking forward to hearing suggestions from you!

Props
-----
* showModal?: state of the modal. Opened if true
* CloseIcon?: component used to customize the close element
* title: title of the modal
* footer: footer of the modal
* rounded?: if true, the modal border is rounded, if not, squared
